fmf/utils.py

```diff
@@ -214,15 +214,23 @@ def filter(filter, data, sensitive=True, regexp=False):
     """
     Return true if provided filter matches given dictionary of values
 
-    Filter supports disjunctive normal form with '|' used for OR, '&'
-    for AND and '-' for negation. Individual values are prefixed with
-    'value:', leading/trailing white-space is stripped. For example::
+    Filter supports disjunctive normal (DNF) form with '|' used for OR,
+    '&' for AND and '-' for negation. Individual values are prefixed
+    with 'value:', leading/trailing white-space is stripped. For
+    example:
 
         tag: Tier1 | tag: Tier2 | tag: Tier3
         category: Sanity, Security & tag: -destructive
 
+    Even though parentheses are only allowed in keys or values they are
```

Comment:
Parenthesis are supported in keys or values? Do you mean like regular characters? 'cause, I think currently there's not support for grouping at all.